kanidmd/src/lib/idm/oauth2.rs

```diff
@@ -499,24 +511,56 @@ impl Oauth2ResourceServersReadTransaction {
             return Err(Oauth2Error::InvalidRequest);
         }
 
-        // We are now GOOD TO GO!
-        // Use this to grant the access token response.
-        let odt_ct = OffsetDateTime::unix_epoch() + ct;
+        // =====================================================
+        // We are now GOOD TO GO! 🥳
+
+        // Later we need to limit this and use refresh tokens.
+        let iat = (OffsetDateTime::unix_epoch() + ct).unix_timestamp();
+        let exp = code_xchg.uat.expiry.unix_timestamp();
 
-        let expires_in = if code_xchg.uat.expiry > odt_ct {
-            // Becomes a duration.
-            (code_xchg.uat.expiry - odt_ct).whole_seconds() as u32
+        // Used in the access token response.
+        let expires_in = if exp > iat {
+            exp - iat
         } else {
             lsecurity!(
                 audit,
                 "User Auth Token has expired before we could publish the oauth2 response"
             );
             return Err(Oauth2Error::AccessDenied);
         };
+        debug_assert!(expires_in.is_positive());
```

Comment:
I don't know what type `exp` and `iat` are but you could probably do something like this to be more idiomatic:

```suggestion
let expires_in = checked_sub(exp, iat).ok_or_else(|| {
    lsecurity!(
        audit,
        "User Auth Token has expired before we could publish the oauth2 response"
    )
    Oauth2Error::AccessDenied
)?;
```

Reply:
expiry and issued at